src/metasploit/utils.py

```diff
@@ -29,4 +29,44 @@ def parseargs():
         print('[-] Error: a password must be specified (-P)\n')
         p.print_help()
         exit(-1)
-    return o
\ No newline at end of file
+    return o
+
+
+def convert_bytes_to_string(bytes_dict: dict):
+    """
+    :param bytes_dict: dictionary, where inner objects are strings, bytes, or collections of those types
+    :return: original dictionary with any bytes values converted to strings
+    """
+    str_data = {}
+    if isinstance(bytes_dict, dict):
+        for key, val in bytes_dict.items():
+            if isinstance(key, bytes):
+                key_temp = key.decode()
+            else:
+                key_temp = key
+            if isinstance(val, list) or isinstance(val, tuple):
+                val_temp = convert_val(val)
+            elif isinstance(val, dict):
+                val_temp = convert_bytes_to_string(val)
+            elif isinstance(val, bytes):
+                val_temp = val.decode()
+            else:
+                val_temp = val
+            str_data[key_temp] = val_temp
+    return str_data
+
+
+def convert_val(bytes_obj: iter):
+    """
+    :param bytes_obj: some object containing either string or byte values, can be collection or just bytes
+    :return: object containing original strings and converted bytes to strings
```

Comment:
Can you clarify the return type a bit here?

Reply:
Added comment